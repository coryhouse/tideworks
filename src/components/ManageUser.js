import React from "react";
import PropTypes from "prop-types";

class ManageUser extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      user: props.user
    };
  }

  componentWillReceiveProps(nextProps) {
    this.setState({ user: nextProps.user });
  }

  handleChange = event => {
    const newUser = Object.assign({}, this.state.user);
    newUser[event.target.name] = event.target.value;
    this.setState({ user: newUser });
  };

  handleSaveClick = event => {
    event.preventDefault();
    this.props.save(this.state.user);
  };

  render() {
    const { id, firstName, lastName } = this.state.user;
    return (
      <form>
        <h1>{id ? "Edit" : "Add"} user</h1>
        First Name:{" "}
        <input
          type="text"
          name="firstName"
          onChange={this.handleChange}
          value={firstName}
        />
        <br />
        Last Name:{" "}
        <input
          type="text"
          name="lastName"
          onChange={this.handleChange}
          value={lastName}
        />
        <br />
        <input type="submit" value="Save" onClick={this.handleSaveClick} />
      </form>
    );
  }
}

ManageUser.propTypes = {
  user: PropTypes.object.isRequired,
  save: PropTypes.func.isRequired
};

export default ManageUser;
