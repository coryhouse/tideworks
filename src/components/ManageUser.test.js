import React from "react";
import ManageUser from "./ManageUser";
import renderer from "react-test-renderer";

it("should render edit header when existing user is provided", () => {
  const user = { id: 1, firstname: "test", lastName: "user" };
  const tree = renderer
    .create(<ManageUser user={user} save={jest.fn} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("should render add header when empty user is provided", () => {
  const user = { id: null, firstname: null, lastName: null };
  const tree = renderer
    .create(<ManageUser user={user} save={jest.fn} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
