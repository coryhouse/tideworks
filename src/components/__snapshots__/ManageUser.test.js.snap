// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render add header when empty user is provided 1`] = `
<form>
  <h1>
    Add
     user
  </h1>
  First Name:
   
  <input
    name="firstName"
    onChange={[Function]}
    type="text"
    value={undefined}
  />
  <br />
  Last Name:
   
  <input
    name="lastName"
    onChange={[Function]}
    type="text"
    value={null}
  />
  <br />
  <input
    onClick={[Function]}
    type="submit"
    value="Save"
  />
</form>
`;

exports[`should render edit header when existing user is provided 1`] = `
<form>
  <h1>
    Edit
     user
  </h1>
  First Name:
   
  <input
    name="firstName"
    onChange={[Function]}
    type="text"
    value={undefined}
  />
  <br />
  Last Name:
   
  <input
    name="lastName"
    onChange={[Function]}
    type="text"
    value="user"
  />
  <br />
  <input
    onClick={[Function]}
    type="submit"
    value="Save"
  />
</form>
`;
